flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/metrics/KafkaSourceReaderMetrics.java

```diff
@@ -0,0 +1,168 @@
+    public static final String KAFKA_SOURCE_READER_METRIC_GROUP = "KafkaSourceReader";
+    public static final String TOPIC_GROUP = "topic";
+    public static final String PARTITION_GROUP = "partition";
+    public static final String CURRENT_OFFSET_METRIC_GAUGE = "currentOffset";
+    public static final String COMMITTED_OFFSET_METRIC_GAUGE = "committedOffset";
+    public static final String COMMITS_SUCCEEDED_METRIC_COUNTER = "commitsSucceeded";
+    public static final String COMMITS_FAILED_METRIC_COUNTER = "commitsFailed";
+    public static final String KAFKA_CONSUMER_METRIC_GROUP = "KafkaConsumer";
+
+    public static final long INITIAL_OFFSET = -1;
+
+    // Metric group for registering Kafka related metrics
+    private final MetricGroup kafkaSourceReaderMetricGroup;
+
+    // Successful / Failed commits counters
+    private final Counter commitsSucceeded;
+    private final Counter commitsFailed;
+
+    // Map for tracking current consuming / committing offsets
+    private final Map<TopicPartition, Offset> offsets = new HashMap<>();
+
+    public KafkaSourceReaderMetrics(MetricGroup parentMetricGroup) {
+        this.kafkaSourceReaderMetricGroup =
+                parentMetricGroup.addGroup(KAFKA_SOURCE_READER_METRIC_GROUP);
+        this.commitsSucceeded =
+                this.kafkaSourceReaderMetricGroup.counter(COMMITS_SUCCEEDED_METRIC_COUNTER);
+        this.commitsFailed =
+                this.kafkaSourceReaderMetricGroup.counter(COMMITS_FAILED_METRIC_COUNTER);
+    }
+
+    /**
+     * Register metrics of KafkaConsumer in Kafka metric group.
+     *
+     * @param kafkaConsumer Kafka consumer used by partition split reader.
+     */
+    @SuppressWarnings("Convert2MethodRef")
+    public void registerKafkaConsumerMetrics(KafkaConsumer<?, ?> kafkaConsumer) {
+        final Map<MetricName, ? extends Metric> kafkaConsumerMetrics = kafkaConsumer.metrics();
+        if (kafkaConsumerMetrics == null) {
+            LOG.warn("Consumer implementation does not support metrics");
+            return;
+        }
+
+        final MetricGroup kafkaConsumerMetricGroup =
+                kafkaSourceReaderMetricGroup.addGroup(KAFKA_CONSUMER_METRIC_GROUP);
+
+        kafkaConsumerMetrics.forEach(
+                (name, metric) ->
+                        kafkaConsumerMetricGroup.gauge(name.name(), () -> metric.metricValue()));
+    }
+
+    /**
+     * Register metric groups for the given {@link TopicPartition}.
+     *
+     * @param tp Registering topic partition
+     */
+    public void registerTopicPartition(TopicPartition tp) {
+        offsets.put(tp, new Offset(INITIAL_OFFSET, INITIAL_OFFSET));
+        registerOffsetMetricsForTopicPartition(tp);
+    }
+
+    /**
+     * Update current consuming offset of the given {@link TopicPartition}.
+     *
+     * @param tp Updating topic partition
+     * @param offset Current consuming offset
+     */
+    public void recordConsumingOffset(TopicPartition tp, long offset) {
```

Comment:
Can we avoid having different naming between the metric name and the method name? For example, it would be good to have `recordCurrentOffset()` rather than `recordConsumingOffset()`. Same applies to `recordCommittingOffset()` which would ideally be `recordCommittedOffset()`.

Reply:
Thanks for reviewing! Fixed in 258483e